src/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkActions.tsx

```diff
@@ -42,20 +47,73 @@ export class ArtworkActions extends React.Component<
     })
   }
 
+  filename() {
+    const { artists, title, date } = this.props.artwork
+    return slugify(
+      compact([artists.map(({ name }) => name).join(", "), title, date]).join(
+        " "
+      )
+    )
+  }
+
+  canDownload(user) {
+    const { is_downloadable } = this.props.artwork
+    return is_downloadable || (user && user.type === "Admin")
+  }
+
+  downloadableImageUrl(user) {
+    const {
+      artwork: { href },
+    } = this.props
+
+    return `${sd.APP_URL}${href}/download/${this.filename()}.jpg`
+  }
+
   render() {
     return (
-      <Container>
-        <SaveButton artwork={this.props.artwork} render={Save(this.props)} />
-        <Spacer mx={0.5} />
-        <ShareButton onClick={this.toggleSharePanel.bind(this)} />
-
-        {this.state.showSharePanel && (
-          <ArtworkSharePanel
-            artwork={this.props.artwork}
-            onClose={this.toggleSharePanel}
-          />
-        )}
-      </Container>
+      <ContextConsumer>
+        {({ user }) => {
+          return (
+            <Container
```

Comment:
The trick is that when the image is downloadable, that last icon and text gets moved all the way to the right with `margin-left: auto;`.

_But_, that has the affect of moving the icons from the center all the way to the _left_, so we use the 50% to bring them back again (if downloadable). Otherwise, we can treat them as normal.